src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -300,12 +351,128 @@ def finalize(self) -> RankBasedMetricResults:  # noqa: D102
             metrics=self.metrics,
             rank_and_candidates=_iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates),
         )
+        if not self.clear_on_finalize:
+            return result
+
         # Clear buffers
         self.ranks.clear()
         self.num_candidates.clear()
 
         return result
 
+    def finalize_multi(self, num: int = 1_000, seed: int = 42) -> Mapping[str, Sequence[float]]:
+        """Bootstrap from :meth:`finalize`.
+
+        :param num:
+            the number of resampling steps
+        :param seed:
+            the random seed.
+
+        :return:
+            a flat dictionary from metric names to list of values
+        """
+        result: DefaultDict[str, List[float]] = defaultdict(list)
+
+        for i in range(num):
+            rank_and_candidates = _iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates)
+            rank_and_candidates = map(functools.partial(RankPack.resample, seed=seed + i), rank_and_candidates)
+            single_result = RankBasedMetricResults.from_ranks(
+                metrics=self.metrics, rank_and_candidates=rank_and_candidates
+            )
+            for k, v in single_result.to_flat_dict().items():
+                result[k].append(v)
+        return result
+
+    def finalize_with_confidence(
+        self,
+        estimator: Union[str, Callable[[Sequence[float]], float]] = np.median,
+        ci: Union[int, str, Callable[[Sequence[float]], float]] = 90,
+        n_boot: int = 1_000,
+        seed: int = 42,
+    ) -> Mapping[str, Tuple[float, float]]:
+        """Finalize result with confidence estimation via bootstrapping.
+
+        Example
```

Comment:
delete this example header, breaks documentation build

Reply:
https://github.com/pykeen/pykeen/pull/1084/commits/b1df776bf075829bf806f90ba6e27907977ede99